### PR TITLE
check errors when writing restart.resume.z

### PR DESCRIPTION
**Pull request by tjhei**

### CODE REVIEW

**Comment by MFraters** [comment]
Looks good! What do you think about also adding an assert to test whether it opened correctly with `f.is_open()`?

File: source/simulator/checkpoint_restart.cc [hunk 1]
```diff
@@ -166,6 +166,13 @@ namespace aspect
           std::ofstream f ((parameters.output_directory + "restart.resume.z").c_str());
           f.write((const char *)compression_header, 4 * sizeof(compression_header[0]));
           f.write((char *)&compressed_data[0], compressed_data_length);
+          f.close();
+
+          if (!f)
```
[the remark this answers is not in the document]
Comment (bangerth) [reply source/simulator/checkpoint_restart.cc]: Hm. I see -- maybe add this as a comment?

**Comment by tjhei** [comment]
This is kind of difficult to test, though. The only thing I could confirm is that the exception triggers correctly if there is not enough disk space to write the file.

**Comment by bangerth** [comment]
OK, thanks for trying. Please add something like this as a comment and then merge yourself.

**Comment by MFraters** [comment]
>> What do you think about also adding an assert to test whether it opened correctly with f.is_open()?
>
>I tried it out: If the open fails, the new Assert also triggers correctly so I would think I don't need a separate check.

Ah, good to know :)

**Pull request merged**